### PR TITLE
#731 OBO Parser: replaced_by

### PR DESCRIPTION
**Pull request by MaxGreil**
#731 
- Modified `OBOFileHandler.java` for triple `replaced_by`
- Added new file `hp.obo` (HPO OBO file) and added new test in `TestOBOFileParsing.java` to test new functionality for HPO OBO file

### CODE REVIEW

File: biojava-ontology/src/test/java/org/biojava/nbio/ontology/TestOboFileParsing.java [hunk 1]
```diff
@@ -37,23 +37,40 @@
...
 
-		Assert.assertTrue(keys.size() > 4000);
-	}
+        Assert.assertTrue(keys.size() > 4000);
+    }
+
+    @Test
+    public void testParsingHPOOBO() throws Exception {
+        OboParser parser = new OboParser();
+        InputStream inStream = parser.getClass().getResourceAsStream("/ontology/hp.obo");
```
Comment (josemduarte) [inline biojava-ontology/src/test/java/org/biojava/nbio/ontology/TestOboFileParsing.java]: If you gzip the hp.obo file, here you just need to use the gzip filter to read it, i.e.
```java
InputStream inStream = new GZIPInputStream(parser.getClass().getResourceAsStream("/ontology/hp.obo"));
```

File: biojava-ontology/src/test/java/org/biojava/nbio/ontology/TestOboFileParsing.java [hunk 2]
```diff
@@ -37,23 +37,40 @@
...
+        Assert.assertNotNull(inStream);
+
+        BufferedReader oboFile = new BufferedReader ( new InputStreamReader ( inStream ) );
+
+        Ontology ontology;
+
+        ontology = parser.parseOBO(oboFile, "Human_phenotype", "the Human Phenotype ontology");
+        Set<Term> keys = ontology.getTerms();
+
+        Assert.assertTrue(keys.size() > 4000);
```
Comment (josemduarte) [inline biojava-ontology/src/test/java/org/biojava/nbio/ontology/TestOboFileParsing.java]: Could you also assert that the new replaced_by field is parsed properly?

**Comment by heuermh** [comment]
> As the new test resource file is quite big I would suggest to gzip it.

It might be better to create an extract of the original file and store it in git uncompressed, that way changes can be tracked with regular line diffs.

**Comment by josemduarte** [comment]
Agree @heuermh , that's a better solution

**Comment by MaxGreil** [comment]
The first appearance of a `replaced_by` field is at line 574. Would it be alright if I then just extract the first 574 lines of the `hp.obo` file?

**Comment by josemduarte** [comment]
@MaxGreil sounds good as long as the file stays consistent and parseable. Thanks!

**Comment by MaxGreil** [comment]
Implemented requested changes:
- Extracted first 574 lines of the original `hp.obo` file

File: biojava-ontology/src/main/java/org/biojava/nbio/ontology/obo/OboFileHandler.java [hunk 3]
```diff
@@ -172,10 +173,14 @@ else if (key.equals(NAME)){
 				Annotation anno = currentTerm.getAnnotation();
 				anno.setProperty(ALT_ID, value);
 			}
-            else if (key.equals(NAMESPACE)){
-                Annotation anno = currentTerm.getAnnotation();
-                anno.setProperty(NAMESPACE, value);
-            }
+			else if (key.equals(NAMESPACE)){
+				Annotation anno = currentTerm.getAnnotation();
```
Comment (heuermh) [inline biojava-ontology/src/main/java/org/biojava/nbio/ontology/obo/OboFileHandler.java]: Minor style fix, could this line be brought outside the `if/else if` block?

File: biojava-ontology/src/test/java/org/biojava/nbio/ontology/TestOboFileParsing.java [hunk 4]
```diff
@@ -34,26 +34,52 @@
 import java.io.InputStream;
 import java.io.InputStreamReader;
 import java.util.Set;
+import java.util.Iterator;
+import org.biojava.nbio.ontology.utils.Annotation;
 
 public class TestOboFileParsing {
 
-	private static final Logger logger = LoggerFactory.getLogger(TestOboFileParsing.class);
+    private static final Logger logger = LoggerFactory.getLogger(TestOboFileParsing.class);
```
Comment (heuermh) [inline biojava-ontology/src/test/java/org/biojava/nbio/ontology/TestOboFileParsing.java]: I'm afraid all of the biojava source should use tabs (I lost that argument a while back :wink:)

**Comment by MaxGreil** [comment]
Fixed style suggestions.

**Comment by MaxGreil** [comment]
Fixed test.

**Comment by MaxGreil** [comment]
@josemduarte there is a new compilation error in the `org.biojava.nbio.structure.tests` and that is weird because I didn't touch any of those files and tests in my last commit

**Comment by josemduarte** [comment]
Yes the tests are still not stable enough. I've restarted the build, hopefully this time it will work.

**Comment by josemduarte** [comment]
Ok finally passing. I'll squash commits and merge. Thanks again!

**Pull request merged**